colossus/src/main/scala/colossus/protocols/memcache/package.scala

```diff
@@ -16,3 +26,150 @@ package object memcache {
   }
 
 }
+
+trait ResponseAdapter[C <: CodecDSL, M[_]] {
```

Comment:
Instead of the usual I, O parameters we use for input/output, I'm using a CodecDSL instead.  There are pros and cons to this, for sure